### PR TITLE
*: adjust tests

### PR DESCRIPTION
**Pull request by disksing**
This PR contains:
1. Add a `Context` argument to `server.Run`, so we can manually stop a server being started.
2. Move some tests to `integration_test` package.
3. Simplify some tests.

### CODE REVIEW

File: cmd/pd-server/main.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ func main() {
 		syscall.SIGTERM,
 		syscall.SIGQUIT)
 
-	if err := svr.Run(); err != nil {
+	if err := svr.Run(sc); err != nil {
```
Comment (siddontang) [inline cmd/pd-server/main.go]: why not pass a Context? So we can cancel the running.

Comment (disksing) [reply cmd/pd-server/main.go]: Good idea.

**Comment by disksing** [comment]
PTAL @siddontang @nolouch

**Comment by siddontang** [comment]
CI failed @disksing

**Comment by disksing** [comment]
@nolouch Please take a look at the last 2 commits.

**Comment by nolouch** [comment]
LGTM

**Pull request merged**